hwilib/devices/bitbox02.py

```diff
@@ -267,31 +294,38 @@ def __init__(self, path: str, password: Optional[str] = None, expert: bool = Fal
                 "The BitBox02 does not accept a passphrase from the host. Please enable the passphrase option and enter the passphrase on the device during unlock."
             )
         super().__init__(path, password=password, expert=expert, chain=chain)
-
-        hid_device = hid.device()
-        hid_device.open_path(path.encode())
-        self.transport = u2fhid.U2FHid(hid_device)
+        simulator = None
+        self.noise_config: BitBoxNoiseConfig = BitBoxNoiseConfig()
+
+        if path != SIMULATOR_PATH:
+            hid_device = hid.device()
+            hid_device.open_path(path.encode())
+            self.transport = u2fhid.U2FHid(hid_device)
+            self.noise_config = CLINoiseConfig()
+        else:
+            simulator = BitBox02Simulator()
+            self.transport = u2fhid.U2FHid(simulator)
         self.device_path = path
 
         # use self.init() to access self.bb02.
         self.bb02: Optional[bitbox02.BitBox02] = None
 
-        self.noise_config: BitBoxNoiseConfig = CLINoiseConfig()
-
     def set_noise_config(self, noise_config: BitBoxNoiseConfig) -> None:
         self.noise_config = noise_config
 
     def init(self, expect_initialized: Optional[bool] = True) -> bitbox02.BitBox02:
         if self.bb02 is not None:
             return self.bb02
 
-        for device_info in devices.get_any_bitbox02s():
+        devs = devices.get_any_bitbox02s()
+        devs.append({'path': SIMULATOR_PATH.encode(), 'serial_number': '', 'product_string': 'bitbox02'})
```

Comment:
Does not feel good to add this regardless of whether we are in simulator mode or not.

Also it's not clean to add it to `devs`, as you don't really need the serial number and product string.

Why is the `for device_info in devs` loop even needed anymore? You don't access `device_info` anymore, so you can just drop the loop and init `bb02` directly, right?

Reply:
The structure was like this from the beginning since all other devices implement it this way : first check if there is a real device, then add also the simulator if `allow_emulators` flag is set.
After our library update and the ability to get info from the device directly, yes, we could drop the loop. It would be a different implementation from the others, but you are right that it would be cleaner.
I will change as you suggested.